ProjFS.Mac/PrjFSKext/PrjFSKext/KauthHandler.cpp

```diff
@@ -539,6 +542,54 @@ static bool ShouldHandleEvent(
     return true;
 }
 
+static bool ShouldHandleFileOpEvent(
+    // In params:
+    vfs_context_t context,
+    const vnode_t vnode,
+    kauth_action_t action,
+
+    // Out params:
+    VirtualizationRoot** root,
+    int* pid)
+{
+    vtype vnodeType = vnode_vtype(vnode);
+    if (ShouldIgnoreVnodeType(vnodeType, vnode))
+    {
+        return false;
+    }
+    
+    // TODO(Mac): We will still want to handle renames into a root, and those vnodes would
+    // not yet have the FileFlags_IsInVirtualizationRoot set
+    uint32_t vnodeFileFlags = ReadVNodeFileFlags(vnode, context);
+    if (!FileFlagsBitIsSet(vnodeFileFlags, FileFlags_IsInVirtualizationRoot))
+    {
+        // This vnode is not part of ANY virtualization root, so exit now before doing any more work.
+        // This gives us a cheap way to avoid adding overhead to IO outside of a virtualization root.      
+        return false;
+    }
+    
+    *root = VirtualizationRoots_FindForVnode(vnode);
+    if (nullptr == *root)
+    {
+        KextLog_FileNote(vnode, "ShouldHandleFileOpEvent(%d): No virtualization root found for file with set flag.", action);
```

Comment:
Just including the `action` value is not all that useful, IMO. Maybe create a helper function for splitting out the action bits like we do for vnodes?

Reply:
Do you mean that it would be more useful to see the name (e.g. KAUTH_FILEOP_CLOSE) than the value?

Unlike the vnode events the file op events are not a bitset:

```
/* Actions */
#define KAUTH_FILEOP_OPEN			1
#define KAUTH_FILEOP_CLOSE			2
#define KAUTH_FILEOP_RENAME			3
#define KAUTH_FILEOP_EXCHANGE	        	4
#define KAUTH_FILEOP_LINK			5
#define KAUTH_FILEOP_EXEC			6
#define KAUTH_FILEOP_DELETE			7
```